libs/ipc/inc/public/tfc/ipc/details/filter.hpp

```diff
@@ -0,0 +1,296 @@
+      process, completion_token, executor);
+}
+}  // namespace detail
+
+template <type_e type, typename value_t>
+struct filter;
+
+template <>
+struct filter<type_e::invert, bool> {
+  using value_t = bool;
+  bool invert{ true };
+  static constexpr type_e type{ type_e::invert };
+
+  auto async_process(value_t const& value, auto&& completion_token) const {
+    auto executor{ asio::get_associated_executor(completion_token) };
+    return asio::async_compose<decltype(completion_token), void(std::expected<value_t, std::error_code>)>(
+        [this, copy = value](auto& self) {
+          if (invert) {
+            self.complete(!copy);
+          } else {
+            self.complete(copy);
+          }
+        },
+        completion_token, executor);
+  }
+
+  struct glaze {
+    using type = filter<type_e::invert, value_t>;
+    static constexpr auto name{ "tfc::ipc::filter::invert" };
+    static constexpr auto value{
+      glz::object("type", &type::type, "Type of filter", "invert", &type::invert, "Invert outputting value")
+    };
+  };
+};
+
+template <>
+struct filter<type_e::timer, bool> {
+  using value_t = bool;
+  std::chrono::milliseconds time_on{ 5000 };
+  std::chrono::milliseconds time_off{ 0 };
+  static constexpr type_e type{ type_e::timer };
+
+  filter() = default;
+  filter(filter&&) noexcept = default;
+  auto operator=(filter&&) noexcept -> filter& = default;
+  filter(filter const& other) {
+    this->time_on = other.time_on;
+    this->time_off = other.time_off;
+  }
+  auto operator=(filter const& other) -> filter& {
+    this->time_on = other.time_on;
+    this->time_off = other.time_off;
+    return *this;
+  }
+
+  auto async_process(value_t const& value, auto&& completion_token) const {
+    auto exe{ asio::get_associated_executor(completion_token) };
+    return asio::async_compose<decltype(completion_token), void(std::expected<value_t, std::error_code>)>(
+        [this, copy = value, first_call = true](auto& self, std::error_code code = {}) mutable {
+          if (code) {
+            fmt::print("TODO DO serious business\n ");
+            self.complete(std::unexpected(code));
+            return;
+          }
+          if (first_call) {
+            first_call = false;
+            if (timer_) {
+              // already waiting need to cancel and return
+              timer_->cancel();  // this will make a recall to this very same lambda with error code
+              timer_ = std::nullopt;
+              return;
+            }
+            auto executor = asio::get_associated_executor(self);
+            timer_ = asio::steady_timer{ executor };
+            if (copy) {
+              timer_->expires_from_now(time_on);
+            } else {
+              timer_->expires_from_now(time_off);
+            }
+            timer_->async_wait(std::move(self));
+          } else {
+            timer_ = std::nullopt;
+            self.complete(copy);
+          }
+        },
+        completion_token, exe);
+  }
```

Comment:
This does not look quite right. Maybe look better into and write some unit tests.